serverlessworkflow/modules/ROOT/pages/getting-started/preparing-environment.adoc

```diff
@@ -0,0 +1,88 @@
+= Environment setup
+
+This guide lists the different ways to set up your environment for {product_name} development. 
+If you are new, start with the minimal one.
+
+[[proc-minimal-local-environment-setup]]
+== Minimal local environment setup
+
+Recomennded steps to setup your local development environment. By completing these steps you are able to
+start the development on your local machine using our guides.
+
+.Procedure
+. Install https://docs.docker.com/engine/install/[Docker] or https://podman.io/docs/installation[Podman].
+. Install https://minikube.sigs.k8s.io/docs/start/[minikube] or https://kind.sigs.k8s.io/docs/user/quick-start/#installation[kind].
+. Install https://kubernetes.io/docs/tasks/tools/[Kubernetes CLI].
+. Install https://knative.dev/docs/install/quickstart-install/[Knative using quickstart]. This will also setup Knative Serving and Eventing for you and the cluster should be running.
+. xref:cloud/operator/install-serverless-operator.adoc[]
+. Install xref:testing-and-troubleshooting/kn-plugin-workflow-overview.adoc[Knative Workflow CLI].
+. Install https://code.visualstudio.com/[Visual Studio Code] with https://marketplace.visualstudio.com/items?itemName=kie-group.swf-vscode-extension[our extension] that simplifies development of workflows by provifing visual aids and auto-complete features.
+
+[[proc-starting-cluster-fo-local-development]]
+== Starting the cluster for local development
+
+If you have used https://knative.dev/docs/install/quickstart-install/[Knative using quickstart] guide, your selected cluster should be running and properly configured to work with our guides.
+
+Please note, that if the knative quickstart procedure is not used, you need to install Knative Serving and Eventing manually. See <<proc-additional-options-for-local-environment>>.
+
++
+.To startup the selected cluster without quickstart, use the following command:
+[tabs]
+====
+Minikube::
++
+--
+.Configure and startup minikube
+[source,shell]
+----
+# Set a driver and container runtime
+minikube config set driver docker/podman
+minikube config set container-runtime docker/podman
+
+# Set cpu and memory
+minikube config set cpus 4
+minikube config set memory 4096
```

Comment:
From the top of my head, since I don't remember exactly the scenario right now, I had issues with that 4096 memory size. Maybe I was working with the prod profile, building a SW, and having data-index and jobs service running, IDK.
Maybe we can increase that number to 6144, or 8192 :thinking: , or, add an inline comment saying that 4096 is a minimal starting point, but if your local machine has enough resources, a higher value is better.